layer_metadata_restorer.go

```diff
@@ -13,29 +13,95 @@ import (
 
 //go:generate mockgen -package testmock -destination testmock/layer_metadata_restorer.go github.com/buildpacks/lifecycle LayerMetadataRestorer
 type LayerMetadataRestorer interface {
-	Restore(buildpacks []buildpack.GroupBuildpack, appMeta platform.LayersMetadata, cacheMeta platform.CacheMetadata) error
+	UseSHAFiles() bool
+	Restore(buildpacks []buildpack.GroupBuildpack, appMeta platform.LayersMetadata, cacheMeta platform.CacheMetadata, layerSHAStore LayerSHAStore) error
 }
 
 type DefaultLayerMetadataRestorer struct {
-	Logger     Logger
-	LayersDir  string
-	SkipLayers bool
+	logger      Logger
+	layersDir   string
+	skipLayers  bool
+	useShaFiles bool
 }
 
-func NewLayerMetadataRestorer(logger Logger, layersDir string, skipLayers bool) LayerMetadataRestorer {
+type LayerSHAStore interface {
+	add(buildpackID, sha string, layer *bpLayer) error
+	get(buildpackID string, layer bpLayer) (string, error)
```

Comment:
This is possible, but is there any particular reason to do this? Making the methods public would allow us to more easily extract a new package if we needed to.

Reply:
Unless you would like to extract them to a new package now, I think that it's better to keep them private for now. If we decide to change it in the future, it's very easy to refactor that.

Reply:
I guess so... I don't really see any precedent for private methods on an interface anywhere else in our code.

Reply:
Eventually I think it makes sense for layer metadata restorer and layer sha store to be part of an internal package. We could do that in a separate PR though.

Reply:
I saw that most interfaces have public methods although they don't really need it.
As I wrote, I don't like it.
I don't mind moving "layer metadata restorer and "layer sha store" to a different package (maybe it's better in a separate PR? I'm not sure) and then change these functions to be public.
Are you ok with that?

Reply:
Made an issue: https://github.com/buildpacks/lifecycle/issues/629